src/profiles.ts

```diff
@@ -0,0 +1,73 @@
+export default class Profiles {
+    private actObjData : Map<string, Array<string>>;
+    private objActData : Map<string, Array<string>>;
+
+    public constructor() {
+        this.actObjData = new Map<string, Array<string>>();
+        this.objActData = new Map<string, Array<string>>();
+    }
+    
+    /*
+        Load the profiles from a JSON string
+    */
+    public loadFromString(s : string) : void {
+        const jsonObj = JSON.parse(s);
+        
+        // Generate data: {key:Actions, value: Array of objects}
+        for (const act in jsonObj) {
+            this.actObjData.set(act, jsonObj[act] as Array<string>);
+        }
+
+        // Generate data: {key:Objects, value: Array of actions}
+        const tmp : {[key:string]: string[]} = {};
+        for (const act in jsonObj) {
+            if (!(jsonObj[act] in tmp)) {
+                tmp[jsonObj[act]] = [];
+            } 
+            tmp[jsonObj[act]].push(act);
+        }
+        for (const obj in tmp) {
+            this.objActData.set(obj, tmp[obj] as Array<string>);
+        }
+
+    }
+
+    public getProfilesJson() : {[key: string]: string[]} {
```

Comment:
You should define `{[key: string]: string[]}` as type.

Reply:
Yes. We should define a type if it's complex.